addons/custom-block-text/userscript.js

```diff
@@ -0,0 +1,89 @@
+import { updateAllBlocks } from "../custom-block-shape/update-all-blocks.js";
+
+export default async function ({ addon, console }) {
+  let currentTextSize = 100;
+
+  const vm = addon.tab.traps.vm;
+  const blocklyInstance = await addon.tab.traps.getBlockly();
+
+  // Handling the CSS from here instead of a userstyle is much more stable, as
+  // there's no code outside of this addon dynamically toggling the styles.
+  // This way, we can clearly control the execution order of style operations.
+  // For example, we always want to call updateAllBlocks() after the styles
+  // were updated according to the user's settings, not before.
+  const fontSizeCss = document.createElement("style");
+  fontSizeCss.textContent = `
+    .blocklyText,
+    .blocklyHtmlInput {
+      font-size: calc(var(--customBlockText-sizeSetting) * 0.12pt) !important;
+    }`;
+  fontSizeCss.disabled = true;
+  document.head.appendChild(fontSizeCss);
+  //
+  const boldCss = document.createElement("style");
+  boldCss.textContent = `
+    .blocklyText,
+    .blocklyHtmlInput {
+      font-weight: bold;
+    }`;
+  boldCss.disabled = true;
+  document.head.appendChild(boldCss);
+  //
+  const textShadowCss = document.createElement("style");
+  textShadowCss.textContent = `
+    .blocklyDraggable > .blocklyText,
+    .blocklyDraggable > g > text {
+      text-shadow: 1px 1px 0 rgba(0, 0, 0, 0.4);
+    }`;
+  textShadowCss.disabled = true;
+  document.head.appendChild(textShadowCss);
+
+  const updateBlockly = () => {
+    blocklyInstance.Field.cacheWidths_ = {}; // Clear text width cache
+    updateAllBlocks(vm);
+  };
+
+  const setFontSize = (wantedSize) => {
+    if (wantedSize !== 100) document.documentElement.style.setProperty("--customBlockText-sizeSetting", wantedSize);
```

Comment:
Nitpick, but could you not use the "customCssVariables" key in `addon.json` to do this automatically?

Reply:
Yeah, but everything here is intentionally handled only by the userscript to make timing 0% dependent on any code outside this addon.
This addon doesn't use userstyles either for this same reason.